darwin/torch/dataset.py

```diff
@@ -182,8 +130,34 @@ def __str__(self):
         return format_string
 
 
+class BaseDataset(Dataset):
+    def __init__(
+        self,
+        dataset_name: str,
+        image_set: str,
+        transforms: Optional = [],
+        client: Optional[Client] = None,
+        **kwargs,
+    ):
+
+        root, split_id = fetch_darwin_dataset(dataset_name, client, **kwargs)
+
+        super(BaseDataset, self).__init__(root, image_set, split_id, transforms)
+        self.classes = []
+
+
 class ClassificationDataset(Dataset):
-    def __init__(self, root: Path, image_set: str, split_id: Optional[str] = None, transforms=None):
+    def __init__(
+        self,
+        dataset_name: str,
+        image_set: str,
+        transforms: Optional = [],
```

Comment:
The type for the `transform` field should be `Optional[List[object]]` (if we're lucky, it could also be `Optional[List[Callable]]` but I haven't tried.

It's always good to try running `mypy` on a branch, to see type related warnings.